WordPress/gutenberg
# Refactor link creation UI to standalone component/file

## What?


Refactors all the link creation components and utilities in `core/navigaiton-link` to a single component/file.


## Why?


This is a step towards goals set out in https://github.com/WordPress/gutenberg/issues/45996#issuecomment-1326255824.

The idea is that we'll need to display the same link ui interface within the Nav block offcanvas experiment as is currently used within the block in the editor canvas.

By extracting everything to a single file it will be easier to see what we have copied/pasted (reused) within the offcanvas.

This also has the additional benefit of cleaning up the code of the Nav Link block.

## How?


Extracts all the Link UI specific stuff to a separate standalone file which can be fed props to configure it for a specific block and link instance.

This should mean we can feed it block data from the offcanvas experiment and it will "just work".

## Testing Instructions





The goal here is to check the link creation UI within the Nav block has not be effected at all. It should remain _exactly_ as it is on `trunk`. Whatever bugs you find here please be sure to cross check them with `trunk`.

- Create some links
- Search for posts/pages ...etc
- Try creating a page directly from within the Link UI (clue: type in a very long page name and it will give you the option to "create").
- Test all the things.

## Screenshots or screencast
